src/hooks/useCart/useCart.tsx

```diff
@@ -107,8 +108,18 @@ const mergeWithCart = (
     );
 };
 
-const hasNoQuota = (quota: Quota): boolean =>
-  quota.remainingQuota.every(item => item.quantity === 0);
+const hasNoQuota = (quota: Quota): boolean => {
+  let [result, hasNegative] = [true, false];
+  for (const item of quota.remainingQuota) {
+    // The negative quantity case should never occur but if it does, we send an error to Sentry and set state to NO_QUOTA
+    if (item.quantity < 0) {
+      Sentry.captureException("Negative Quota Received");
+      item.quantity = 0;
+      hasNegative = true;
+    } else result = result && item.quantity === 0;
+  }
+  return result || hasNegative;
+};
```

Comment:
Couple of guiding questions to improve this:

1. Given `hasNoQuota` returns a boolean, will a consumer know that it actually has side effects (such as changing item quantity to 0)? Why should this function have side effects? Could there be a better approach?
2. How might we make use of existing TS constructs to further simplify the code? Can we make use of existing TS array functions (e.g. `every`) to help achieve the same outcome?

Let me know if you have questions!